cmd/k8s-operator/deploy/chart/templates/deployment.yaml

```
@@ -25,6 +24,10 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
     spec:
+      {{- if .Values.operatorConfig.hostNetwork.enabled }}
+      hostNetwork: true
+      dnsPolicy: {{ .Values.operatorConfig.hostNetwork.dnsPolicy }}
```

To illustrate my other comment, this suggestion plus the one below:

```suggestion
      dnsPolicy: ClusterFirstWithHostNet
```

---

is there additional benefit to not letting this be configurable? I'm trying to understand why we'd hardcode it. This is likely the default most people will use but I'd like to figure out why we'd prevent people configuring it.

---

@jaxxstorm generally we try to not add extra config knobs to any API unless there is a known user need for it.

Anything that we make configurable today, we'll have to support 'forever' with the assumption that any user might have configured it to whatever value. If we hardcode it today, it gives us freedom to make changes in the future + if in future there is a need to make it configurable, we can do so.

On top of that, if we make something configurable we should also be able to understand/explain to users why they would want to configure it.

---

Okay, that helps. I've updated.